hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcClient.java

```diff
@@ -478,8 +498,13 @@ public Object invokeMethod(
       rpcMonitor.proxyOp();
     }
     boolean failover = false;
+    boolean tryActive = false;
```

Comment:
Do we need tryActive? The observers will be sorted ahead of the active ones, so we should only use them if we have gone through all of the observers.

Reply:
We don't need an extra variable. I now set `shouldUseObserver` to false if there is a RetryException from the Observer.